### PR TITLE
[ios] Fixed XCode warnings

### PR DESCRIPTION
**Pull request by biodranik**
There are still too much warnings, need to clear them all some day.

### CODE REVIEW

File: routing/ruler_router.cpp [hunk 1]
```diff
@@ -66,7 +66,7 @@ RouterResultCode RulerRouter::CalculateRoute(Checkpoints const & checkpoints,
     return geometry::PointWithAltitude(p, 0 /* altitude */);
   };
 
-  for (size_t i = 0; i < count; ++i)
+  for (uint32_t i = 0; i < count; ++i)
```
[the remark this answers is not in the document]
Comment (biodranik) [reply routing/ruler_router.cpp:69]: TurnItem m_index is uint32_t

**Review by vng — APPROVED** [review]

File: iphone/Maps/Core/EventListening/ListenerContainer.swift [hunk 2]
```diff
@@ -3,15 +3,15 @@ import Foundation
 // TODO: define directly T as AnyObject after Swift version update
 final class ListenerContainer<T> {
   // MARK: - WeakWrapper for listeners
-  private class WeakWrapper<T> {
+  private class WeakWrapper<TT> {
```
Comment (kirylkaveryn) [inline iphone/Maps/Core/EventListening/ListenerContainer.swift:6]: Maybe use not `TT`, but `H` or `U` to not confuse between T and TT?  Generic type placeholders in Swift usually don't use double letterings...

File: iphone/Maps/UI/BottomMenu/Menu/Cells/BottomMenuLayersCell.xib [hunk 3]
```diff
@@ -1,9 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<document type="com.apple.InterfaceBuilder3.CocoaTouch.XIB" version="3.0" toolsVersion="22155" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES">
+<document type="com.apple.InterfaceBuilder3.CocoaTouch.XIB" version="3.0" toolsVersion="32700.99.1234" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES">
     <device id="ipad9_7" orientation="landscape" layout="fullscreen" appearance="light"/>
     <dependencies>
         <deployment identifier="iOS"/>
-        <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="22131"/>
+        <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="22684"/>
         <capability name="Safe area layout guides" minToolsVersion="9.0"/>
```
Comment (kirylkaveryn) [inline iphone/Maps/UI/BottomMenu/Menu/Cells/BottomMenuLayersCell.xib:7]: I don't like XIB and Storyboards because of these huge changes...

**Review by kirylkaveryn — APPROVED** [review]

File: iphone/Maps/Core/EventListening/ListenerContainer.swift (hunk 2 above)
Comment (biodranik) [reply iphone/Maps/Core/EventListening/ListenerContainer.swift:6]: Doesn't TT imply that it's inside the T type? :)

This is not critical at all, there is a TODO above, maybe a better fix will be to implement it and remove this internal type completely.

File: iphone/Maps/UI/BottomMenu/Menu/Cells/BottomMenuLayersCell.xib (hunk 3 above)
Comment (biodranik) [reply iphone/Maps/UI/BottomMenu/Menu/Cells/BottomMenuLayersCell.xib:7]: Nobody likes it...

**Pull request merged**